qiskit/transpiler/passes/basis/unroll_custom_definitions.py

```diff
@@ -67,11 +73,22 @@ def run(self, dag):
             if dag.has_calibration_for(node):
                 continue
 
-            if node.name in device_insts or self._equiv_lib.has_entry(node.op):
-                if isinstance(node.op, ControlledGate) and node.op._open_ctrl:
-                    pass
-                else:
-                    continue
+            if self._target is not None:
+                inst_supported = self._target.instruction_supported(
+                    operation_name=node.op.name, qargs=tuple(qubit_mapping[x] for x in node.qargs)
+                )
+                if inst_supported or self._equiv_lib.has_entry(node.op):
+                    if isinstance(node.op, ControlledGate) and node.op._open_ctrl:
+                        pass
+                    else:
+                        continue
+
+            else:
+                if node.name in device_insts or self._equiv_lib.has_entry(node.op):
+                    if isinstance(node.op, ControlledGate) and node.op._open_ctrl:
+                        pass
+                    else:
+                        continue
```

Comment:
```suggestion
            controlled_gate_open_ctrl = isinstance(node.op, ControlledGate) and node.op._open_ctrl
            if not controlled_gate_open_ctrl:
                inst_supported = self._target.instruction_supported(
                    operation_name=node.op.name, qargs=tuple(qubit_mapping[x] for x in node.qargs)
                ) if self._target is not None else node.name in device_insts

                if inst_supported or self._equiv_lib.has_entry(node.op):
                    continue
```

Untested (not sure if Python will be happy with `inst_supported` statement...), but hopefully something along these lines will improve readability.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8784/commits/d87ce39027844b16a3a449a6ed97ca6fdb2f87c1